handlers/lookup.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 OpenFaaS Author(s)
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+type FunctionLookup struct {
+}
+
+func (l *FunctionLookup) Resolve(name string) (url.URL, error) {
+	namespace := "openfaas-fn"
+
+	urlRes, errRes := url.Parse(fmt.Sprintf("http://%s.%s:8080", name, namespace))
+
+	return *urlRes, errRes
+}
+
+func NewFunctionLookup() http.HandlerFunc {
```

Comment:
Why do we need this noop handlerfunc factory?

Reply:
You tell me.. I'm trying to implement the proxy you built and it seemed to be required. This fixes the issue, but if you can do your own PR it would be better.

Reply:
@LucasRoesler, I'm happy for you to take over this PR give that you built the proxy API in use here.